public/app/features/alerting/unified/api/alertRuleApi.ts

```diff
@@ -225,24 +225,63 @@ export const alertRuleApi = alertingApi.injectEndpoints({
     // TODO This should be probably a separate ruler API file
     getRuleGroupForNamespace: build.query<
       RulerRuleGroupDTO,
-      { rulerConfig: RulerDataSourceConfig; namespace: string; group: string }
+      WithRequestOptions<{ rulerConfig: RulerDataSourceConfig; namespace: string; group: string }>
     >({
-      query: ({ rulerConfig, namespace, group }) => {
+      query: ({ rulerConfig, namespace, group, requestOptions }) => {
         const { path, params } = rulerUrlBuilder(rulerConfig).namespaceGroup(namespace, group);
-        return { url: path, params };
+        return withRequestOptions({ url: path, params }, requestOptions);
       },
-      providesTags: ['CombinedAlertRule'],
+      providesTags: (_result, _error, { namespace, group }) => [
+        {
+          type: 'RuleGroup',
+          id: `${namespace}/${group}`,
+        },
+        { type: 'RuleNamespace', id: namespace },
+      ],
     }),
 
     deleteRuleGroupFromNamespace: build.mutation<
       RulerRuleGroupDTO,
-      { rulerConfig: RulerDataSourceConfig; namespace: string; group: string }
+      WithRequestOptions<{ rulerConfig: RulerDataSourceConfig; namespace: string; group: string }>
     >({
-      query: ({ rulerConfig, namespace, group }) => {
+      query: ({ rulerConfig, namespace, group, requestOptions }) => {
         const { path, params } = rulerUrlBuilder(rulerConfig).namespaceGroup(namespace, group);
-        return { url: path, params, method: 'DELETE' };
+        return withRequestOptions({ url: path, params, method: 'DELETE' }, requestOptions, {
+          successMessage: 'Successfully deleted rule group',
+        });
+      },
+      invalidatesTags: (_result, _error, { namespace, group }) => [
+        {
+          type: 'RuleGroup',
+          id: `${namespace}/${group}`,
+        },
+        { type: 'RuleNamespace', id: namespace },
+      ],
+    }),
+
+    upsertRuleGroupForNamespace: build.mutation<
+      AlertGroupUpdated,
+      WithRequestOptions<{
+        rulerConfig: RulerDataSourceConfig;
+        namespace: string;
+        payload: PostableRulerRuleGroupDTO;
+      }>
+    >({
+      query: ({ payload, namespace, rulerConfig, requestOptions }) => {
+        const { path, params } = rulerUrlBuilder(rulerConfig).namespace(namespace);
+
+        return withRequestOptions(
+          {
+            url: path,
+            params,
+            data: payload,
+            method: 'POST',
+          },
+          requestOptions,
+          { successMessage: 'Successfully updated rule group' }
```

Comment:
(Slightly duplication - we have this translated elsewhere, but not here. Perhaps we could use the `t` function within here? Not essential though right now if that turns out to be problematic for some reason)

Reply:
Oh I had intended to translate all of them, let me fix that